### DISCUSSION TITLE
Generate types and queries dynamically on the fly

### DISCUSSION THREAD

**Discussion by khanakia** [strawberry-graphql/strawberry#3225] [Q&A]
I am building a CMS and each user has different Models and Field Names. 

So how do i generate the the Types and fields dynamically?

**Comment by patrick91** [comment]
@khanakia you can use `create_type` for that: https://strawberry.rocks/docs/guides/tools#create_type

**Reply by khanakia** [reply]
Getting this error ![Screenshot 2023-11-14 at 9 27 44 PM](https://github.com/strawberry-graphql/strawberry/assets/2174170/ac7dd751-9177-4233-9f1e-c49379490fc2)

**Reply by patrick91** [reply]
did you run the code? that error is only pylance, but it should still work, and you can use `# type: ignore` to disable the error

**Reply by khanakia** [reply]
thanks, it worked.

Will you be able to provide some example please how do I achieve this query `post: Post!`

How do I create the `POST` object dynamically as the current example only have `-> str` examples.

**Reply by patrick91** [reply]
query is just like any other type, so you make a post one you can do the same code pretty much

```python
import strawberry
from strawberry.tools import create_type

@strawberry.field
def title(info) -> str:
    return "World"

Post = create_type("Post", [title])

def get_post(info) -> Post:
    return Post()
 
post = strawberry.field(name="post", resolver=get_post)

Query = create_type("Query", [post])

schema = strawberry.Schema(query=Query)
```

**Reply by patrick91** [reply]
see it in action here: https://play.strawberry.rocks/?gist=028cb4560b6769d08cdec5e9d1ec940d

**Reply by khanakia** [reply]
Cool 😎  Thanks for building such a great library.

One last question is it okay to ignore these pylance warnings ?

![Screenshot 2023-11-14 at 11 27 33 PM](https://github.com/strawberry-graphql/strawberry/assets/2174170/4d883169-911e-47c5-9414-805e3fd6731c)

**Reply by patrick91** [reply]
yes, but I wonder if we could make the typings better 😊
